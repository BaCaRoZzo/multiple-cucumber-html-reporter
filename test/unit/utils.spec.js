const path = require('path');
const fs = require('fs-extra');
const {findJsonFiles, formatToLocalIso} = require('../../lib/utils');

jest.mock('path');
jest.mock('fs-extra');
jest.mock('moment', () => () => ({format: () => '2020/02/17 16:27:43'}));

afterEach(() => jest.resetAllMocks());

describe('findJsonFiles', () => {
    it('should be able to find all json files', () => {
        const resolveMock = '/resolveMock';
        const files = ['1.json', '2.json', '3.json', '4.txt', '5.doc', '6.json'];
        const resolveSpy = path.resolve.mockReturnValueOnce(resolveMock);
        const readdirSyncSpy = fs.readdirSync.mockReturnValueOnce(files);
        const joinSpy = path.join
            .mockReturnValueOnce(`${resolveMock}/${files[0]}`)
            .mockReturnValueOnce(`${resolveMock}/${files[1]}`)
            .mockReturnValueOnce(`${resolveMock}/${files[2]}`)
            .mockReturnValueOnce(`${resolveMock}/${files[5]}`);

        expect(findJsonFiles('./test/unit/data/collect-jsons')).toMatchSnapshot();
        expect(resolveSpy).toHaveBeenCalledTimes(1);
        expect(readdirSyncSpy).toHaveBeenCalledTimes(1);
        expect(joinSpy).toHaveBeenCalledTimes(4);
    });

    it('should throw an error when the json folder does not exist', () => {
        const resolveMock = '/resolveMock';
        path.resolve.mockReturnValueOnce(resolveMock);
        const readdirSyncSpy = fs.readdirSync.mockReturnValueOnce(null);
        const joinSpy = path.join

        expect(()=> findJsonFiles(resolveMock))
            .toThrow(new Error(`There were issues reading JSON-files from '${resolveMock}'.`));
        expect(readdirSyncSpy).toHaveBeenCalledTimes(1);
        expect(joinSpy).toHaveBeenCalledTimes(0);
    });

    it('should return an empty array when no json files could be found', () => {
        const resolveMock = '/resolveMock';
        const files = [];
        const resolveSpy = path.resolve.mockReturnValueOnce(resolveMock);
        const readdirSyncSpy = fs.readdirSync.mockReturnValueOnce(files);
        const joinSpy = path.join

        expect(findJsonFiles('./no-jsons')).toMatchSnapshot();
        expect(readdirSyncSpy).toHaveBeenCalledTimes(1);
        expect(joinSpy).toHaveBeenCalledTimes(0);
    });
});

describe('formatToLocalIso', () => {
    it('should be able to format time to local ISO', () => {
        expect(formatToLocalIso('2020-02-17T15:27:43.843Z')).toMatchSnapshot();
    });
});
