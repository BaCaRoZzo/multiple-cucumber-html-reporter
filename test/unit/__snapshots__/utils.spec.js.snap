// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`findJsonFiles should be able to find all json files 1`] = `
Array [
  "/resolveMock/1.json",
  "/resolveMock/2.json",
  "/resolveMock/3.json",
  "/resolveMock/6.json",
]
`;

exports[`findJsonFiles should return an empty array when no json files could be found 1`] = `Array []`;

exports[`formatToLocalIso should be able to format time to local ISO 1`] = `"2020/02/17 16:27:43"`;
